calculator.js

```diff
@@ -1 +1,180 @@
-console.log('start')
\ No newline at end of file
+const numbers = document.querySelectorAll('.number')
+const mainResult = document.querySelector('.result')
+const operators = document.querySelectorAll('.operator')
+const equal = document.querySelectorAll('.equal')
+const deleteEqual = document.querySelectorAll('.deleteEqual')
+const dot = document.querySelectorAll('.dot')
+const deleteLast = document.querySelectorAll('.deleteLast')
+
+let result = ''
+let number1 = ''
+let number2 = ''
+let operator = undefined
+
+count = () => {
+  switch (String(operator)) {
+    case '+': result = add(Number(number1), Number(number2))
+      break;
+    case '-': result = subtract(Number(number1), Number(number2))
+      break;
+    case '/': result = divide(Number(number1), Number(number2))
+      break;
+    case '*': result = multiply(Number(number1), Number(number2))
+      break;
+    default:
+      return
+  }
+}
+add = (a, b) => {
+  return a + b
+}
+subtract = (a, b) => {
+  return a - b
+}
+divide = (a, b) => {
+  return a / b
+}
+multiply = (a, b) => {
+  return a * b
+}
+concat = (e) => {
+  result = result.toString() + e.toString()
+  render()
+}
+render = () => {
+  mainResult.innerText = result
+}
+
+check = () => {
+  if (result !== '' && result !== '-' && number1 === '' && number2 === '') {
+    number1 = result
+    result = 0
+    render()
+    result = ''
+    return
+  }
+  if (result !== '' && number1 !== '' && number2 === '') {
+    number2 = result
+    count()
+    render()
+    number1 = result
+    number2 = ''
+    result = ''
+    operator = undefined
+    return
+  }
+}
+
+changeOperator = (e) => {
+  if (number1 === '' && result === '' && operator === undefined) {
+    result = '-'
+    render()
+    return
+  }
+  if (number1 !== '' && result === '' && operator === undefined) {
+    operator = e
+    result = 0
+    render()
+    result = ''
+    return
+  }
+  if (number1 !== '' && result === '' && operator === '-') {
+    operator = e
+    result = '-'
+    render()
+    return
+  }
+  if (number1 !== '' && result === '' && operator !== undefined && operator !== '-') {
+    result = '-'
+    render()
+    return
+  }
+  operator = e
+  check()
+}
+
+numbers.forEach((numberFromButtonClick) => {
+  numberFromButtonClick.addEventListener(
+    'click',
+    () => {
+      if (numberFromButtonClick.innerText === String(0)) {
+        if (result === String(0)) {
+          return
+        }
+      }
+      concat(numberFromButtonClick.innerText)
+    }
+  )
+})
+
+operators.forEach((e) => {
```

Comment:
The param name misleading again, it is not e like event it is button or element. If you must do short vars go with `el` not `e` but it is always wise to be more explicit and go with longer ones.